engine/src/main/scala/cromwell/engine/workflow/ShadowWorkflowActor.scala

```diff
@@ -164,4 +169,39 @@ class ShadowWorkflowActor(workflowId: WorkflowId, startMode: StartMode, wdlSourc
       log.warning(s"$tag received an unhandled message: $unhandledMessage")
       stay
   }
+
+  private def assignCallsToBackends(calls: Seq[Call], workflowOptions: WorkflowOptions): Map[Call, String] = {
+    val backendKey = "backend"
+    // ------------------------------------------------------------------//
+    // TODO: This should be obtained from the backend config factory. Part of issue #649
+    // This is currently a stub. Should be removed as part of #649 (?)
+    val listOfPluggedInBackends = Seq("local", "sge", "jes", "htcondor")
+    val defaultBackend = "local"
+    // ------------------------------------------------------------------//
+
+    import scala.collection.breakOut
+    val backendFromOptions = workflowOptions.get(backendKey)
+    val callAssignmentMap = backendFromOptions match {
+      case Success(backendName) =>
+        // Use this backend for all the calls in the workflow
+        calls.map(_ -> backendName)(breakOut): Map[Call, String]
+      case Failure(reason) =>
+        log.warning("Backend was not defined in the workflow options, trying to runtime-attributes based per task backend allocation.", reason)
+        calls.map { call =>
```

Comment:
There looks to be multiple sub-functions which could exist instead of this full closure